pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -346,18 +346,64 @@ func (d *DeploymentController) configureIstioGateway(log *istiolog.Scope, gw gat
 		serviceType = corev1.ServiceType(o)
 	}
 
+	// TODO: Codify this API (i.e how to know if a specific gateway is an Istio waypoint gateway)
+	isWaypointGateway := strings.Contains(string(gw.Spec.GatewayClassName), "waypoint")
+
+	// Default the network label for waypoints if not explicitly set in gateway's labels
+	if _, ok := gw.GetLabels()["topology.istio.io/network"]; !ok && isWaypointGateway {
```

Comment:
why only for waypoint? before we added it always?

Reply:
This came from a merge conflict; I'm not sure which PR brought this in

Reply:
This logic replaces this logic: https://github.com/istio/istio/pull/47229/files#diff-bd23661d98745c4cf580be1d1ee08e08572867f1d8ca535883c06aaa591b5760L38